gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/QueryResultsImpl.java

```diff
@@ -86,8 +89,8 @@ protected T computeNext() {
     if (!entityResultPbIter.hasNext()) {
       return endOfData();
     }
-    DatastoreV1.EntityResult entityResultPb = entityResultPbIter.next();
-    //cursor = entityResultPb.getCursor(); // only available in v1beta3
+    com.google.datastore.v1beta3.EntityResult entityResultPb = entityResultPbIter.next();
+    cursor = entityResultPb.getCursor();
```

Comment:
Ah sorry. Yes, they should always have a value. The `end_cursor` will as well. 

Yes in general the last result cursor will be the same as the end cursor. However it's possible that the Datastore could make progress in the backend and then return a cursor that is further along than the last result cursor.